### ISSUE TITLE
[Feature]: Enable HTTP 2.0 on Nginx

### ISSUE THREAD

**Issue by sumitsum** [appsmithorg/appsmith#27919]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Currently, Appsmith application only seems to support HTTP 1.x protocol. APMU (please investigate further) - the API server is already HTTP 2 enabled, which means the issue resides with Nginx not being HTTP 2 enabled. Please check out the below references for more details: 
[Reference1](https://theappsmith.slack.com/archives/C024GUDM0LT/p1696840752702209)
[Reference2](https://theappsmith.slack.com/archives/C0341RERY4R/p1696423706878489) 

<img width="1005" alt="Screenshot 2023-10-12 at 6 03 37 PM" src="https://github.com/appsmithorg/appsmith/assets/1757421/994fe2b3-dd51-4eb8-8e8e-7fbaf12d179f">

Update:
looks like there might be another issue here - we need to enable http2 on webclient as well. Tracking it [here](https://github.com/appsmithorg/appsmith/issues/28226). 

### Why should this be worked on?

Enable HTTP 2.0

On cloud setup:
There are 6 api calls that happen in http 1.0 and 2 get blocked right now on our cloud infrastructure. With HTTP 2.0, all the api calls, i.e 8 API calls should get executed in parallel now.

**Comment by Peterkrol12** [comment]
We encountered this issue during one of our use cases. We have a service running in ECS behind an ALB. This ALB is configured with an `HTTPS:443` listener with an HTTP2 target group. Calling an endpoint on our service returns a [464 status code](https://http.dev/464), which indicates an incompatible protocol. Therefore, this call does not exceed and we can not use our service behind this HTTP2 enabled ALB.

**Comment by sharat87** [comment]
@Peterkrol12, can you confirm if I'm understanding this right. You're trying to build an API action on Appsmith, that hits an API running on ECS behind an ALB, that's running on HTTP2. Is that correct? In that case, you're looking for a solution to https://github.com/appsmithorg/appsmith/issues/28226 instead.

This issue is about having Appsmith itself be able to run with HTTP2.

**Comment by rajatagrawal** [comment]
Modern browsers don't support http2 without SSL. So having a docker image with HTTP2 isn't working. Closing this issue as this not possible technically.

**Comment by rajatagrawal** [comment]
Had a follow up with @sumitsum and we found out that we can use H2C instead of HTTP 2.0 and we should have the same benefits as HTTP 2.0 without SSL. This would be applicable to docker images running simply on an ip address without SSL.

**Comment by sumitsum** [comment]
Closing for now it seems that none of the major browsers support h2c protocol. 
https://http2.github.io/faq/
```
However, some implementations have stated that they will only support HTTP/2 when it is used over an encrypted connection, and currently no browser supports HTTP/2 unencrypted.
```

https://bugzilla.mozilla.org/show_bug.cgi?id=1418832
